File: test-project/src/main/java/TestClass.java
public class TestClass {
    public static void main(String[] args) {
        System.out.println("Starting application...");
        
        #if DEBUG
        System.out.println("Debug mode is enabled");
        System.out.println("Detailed logging active");
        #else
        System.out.println("Production mode");
        #endif
        
        #if VERSION==1.0
        System.out.println("Version 1.0 features enabled");
        #elif VERSION==2.0
        System.out.println("Version 2.0 features enabled");
        #else
        System.out.println("Unknown version");
        #endif

        System.out.println("Application initialized");
    }
}

File: test-project/src/main/kotlin/TestClass.kt
fun main(args: Array<String>) {
	println("Starting application...")

	#if DEBUG == true
	println("Debug mode is enabled");
	println("Detailed logging active");
	#else
	println("Production mode");
	#endif

	#if VERSION == "1.0"
	println("Version 1.0 features enabled");
	#elif VERSION == "2.0"
	println("Version 2.0 features enabled");
	#else
	println("Unknown version");
	#endif

	println("Application initialized");
}